meta-resin-common/recipes-support/resin-vars/resin-vars/os-sshkeys

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+#
+# Copyright 2018 Resinio Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+tool=$(basename "$0")
+custom_ssh_public_keys=".os.sshKeys"
+authorized_keys_path=/home/root/.ssh/authorized_keys_local
+authorized_keys_perm="600"
+
+log () {
+	echo "$tool: $1"
+}
+
+# shellcheck disable=SC2153
+if [ -n "$AUTHORIZED_KEYS_PATH" ]; then
+	authorized_keys_path="$AUTHORIZED_KEYS_PATH"
+fi
+
+if [ -z "$CONFIG_PATH" ]; then
+	. /usr/sbin/resin-vars
+	[ -z "$CONFIG_PATH" ] && { log "No config.json found."; exit 1; }
+fi
+
+ssh_keys=$(jq -r "select(${custom_ssh_public_keys} != null) | \
+	${custom_ssh_public_keys}[] //empty" "$CONFIG_PATH")
+
+if [ -n "$ssh_keys" ]; then
+	log "Setting up custom SSH public keys..."
+	authorized_keys=""
+	IFS="
+"
+
+	for key in $ssh_keys; do
+		if [ -z "$authorized_keys" ]; then
+			authorized_keys="$(printf '%s' "${key}")"
+		else
+			authorized_keys="$(printf '%s\n%s' "${authorized_keys}" \
+				"${key}")"
+		fi
+	done
+	unset IFS
+
```

Comment:
For my own understanding, @agherzan why do you build this `$authorized_keys` variable above: I would think it would still have the same value as the `$ssh_keys` variable... So why not use directly `$ssh_keys` below?

Or maybe it has something to do with platform specific EOL normalization like windows `\r\n` EOL chars?

Reply:
`ssh_key` is what you get from the `config.json` while `authorized_keys` is what you have in the file. If they don't match at any point the authorized_keys_path file is refreshed. This could happen because the user changed their keys in config.json for example.

Reply:
Are you sure? `authorized_keys` variable seem to be created from the `ssh_keys` in the for loop, but why not use `ssh_keys` directly (we could compare `ssh_keys` value with the content of the `authorized_keys_path` file directly and replace its content if different)

Reply:
@bbinet right. Checking the code a little closer I realized what you are talking about. If a remember correctly, It was due to some EOL issues I ran into.